test/fakeintake/api/api.go

```diff
@@ -13,10 +13,20 @@ type Payload struct {
 	Encoding  string    `json:"encoding"`
 }
 
-type APIFakeIntakePayloadsGETResponse struct {
+type ParsedPayload struct {
```

Comment:
Instead of having two structs (`Payload` and `ParsedPayload`) you can have  `type Payload[T any] struct`.
Same comment for `APIFakeIntakePayloadsGETResponse`.